src/lib/services/generation/ChapterBatchService.ts

```
@@ -0,0 +1,184 @@
+    callbacks: ChapterBatchCallbacks,
+  ): Promise<ChapterBatchResult> {
+    const boundaries = planChapterBoundaries(
+      input.entries,
+      input.tokenThreshold,
+      input.chapterBuffer,
+      input.startIndex,
+    )
+
+    log('Batch planned', { boundaryCount: boundaries.length, startIndex: input.startIndex })
+
+    let chapters: Chapter[] = []
```

ESLint

[error] 'chapters' is never reassigned. Use 'const' instead.

(prefer-const)